src/triage/component/architect/feature_group_mixer.py

```diff
@@ -31,6 +32,23 @@ def leave_one_out(feature_groups):
         results.append(feature_dict)
     return results
 
+def all_combinations(feature_groups):
+    """Return all combinations of groups, excluding repeated groups
+
+    Args:
+        feature_groups (list) The feature groups to apply the strategy to
+
+    Returns: A list of feature dicts
+    """
+    results = []
+    for number_feature_groups in range(0, len(feature_groups) + 1):
```

Comment:
the more explicit version is fine, but this call to `range` _can_ just be:

```suggestion
    for number_feature_groups in range(len(feature_groups) + 1):
```